src/Components/WebAssembly/WebAssembly/src/Hosting/WebAssemblyHost.cs

```diff
@@ -139,7 +139,7 @@ internal async Task RunAsyncCore(CancellationToken cancellationToken, WebAssembl
         manager.SetPlatformRenderMode(RenderMode.InteractiveWebAssembly);
         await manager.RestoreStateAsync(store);
 
-        if (MetadataUpdater.IsSupported)
+        if (MetadataUpdater.IsSupported && Environment.GetEnvironmentVariable("__BLAZOR_WEBASSEMBLY_LEGACY_HOTRELOAD") != "false")
```

Comment:
Leave built-in HotReload in place, just disable it when nuget package is referenced. The plan is to remove it later

Reply:
I would expect that ILLink will understand the condition, but I'll check

Reply:
Confirmed, the whole namespace gets trimmed
<img width="503" height="227" alt="image" src="https://github.com/user-attachments/assets/85802e00-7264-4ed3-99d0-5e3105301168" />

Reply:
When is later, once the SDK change has propagated or post 10.0